darwin/dataset/remote_dataset_v2.py

```diff
@@ -642,3 +654,113 @@ def register(
                 print(f"  - {item}")
         print(f"Reistration complete. Check your items in the dataset: {self.slug}")
         return results
+
+    def register_multi_slotted(
+        self,
+        object_store: ObjectStore,
+        storage_keys: Dict[str, List[str]],
+        fps: Optional[Union[str, float]] = None,
+        multi_planar_view: bool = False,
+        preserve_folders: bool = False,
+    ) -> Dict[str, List[str]]:
+        """
+        Register files in the dataset in multiple slots.
+
+        Parameters
+        ----------
+        object_store : ObjectStore
+            Object store to use for the registration.
+        storage_keys : Dict[str, List[str]
+            Storage keys to register. The keys are the item names and the values are lists of storage keys.
+        fps : Optional[str], default: None
+            When the uploading file is a video, specify its framerate.
+        multi_planar_view : bool, default: False
+            Uses multiplanar view when uploading files.
+        preserve_folders : bool, default: False
+            Specify whether or not to preserve folder paths when uploading
+
+        Returns
+        -------
+        Dict[str, List[str]]
+            A dictionary with the list of registered files.
+
+        Raises
+        ------
+        ValueError
+            If ``storage_keys`` is not a dictionary with keys as item names and values as lists of storage keys.
+        TypeError
+            If the file type is not supported.
+        """
+        if not isinstance(storage_keys, dict) or not all(
+            isinstance(v, list) and all(isinstance(i, str) for i in v)
+            for v in storage_keys.values()
+        ):
+            raise ValueError(
+                "storage_keys must be a dictionary with keys as item names and values as lists of storage keys."
+            )
```

Comment:
pydantic models could be very efficient here! - but i guess we only use pydantic in darwin-py 2.0 🤔

Reply:
This is a really good point. We do, but I don't see a problem with using models here

We could do something like:

```
class StorageKeysModel(BaseModel):
    storage_keys: Dict[str, List[str]] = Field(..., description="A dictionary with keys as item names and values as lists of storage keys.")
```

and then:

```
try:
    validated_input = StorageKeysModel(storage_keys=storage_keys)
except ValidationError as e:
    raise ValueError(e)
```

What do you think?

Reply:
Perfect, that'd work! 🙏